omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -146,18 +158,30 @@ function handleMonthlyPeriodChange() {
     selectedPeriod = selectedValue.format('YYYYMM');
 }
 
-function handleYearlyPeriodChange() {
-    const yearlyPicker = jQuery('#yearlyPicker');
+function handleFinancialYearPeriodChange() {
+    const yearlyPicker = jQuery('#financialYearPicker');
     let selectedYear = yearlyPicker.val();
     const currentYear = moment().year();
+    const currentMonth = moment().month();
     // Set back to the maximum possible year if the user entered a wrong value
     if (selectedYear >= currentYear) {
-        yearlyPicker.val(currentYear - 1);
-        selectedYear = currentYear - 1;
+        if (currentMonth > moment().month(financialYearMonth).format("M")) {
+            yearlyPicker.val(currentYear - 1);
+            selectedYear = currentYear - 1;
+        } else {
+            yearlyPicker.val(currentYear - 2);
+            selectedYear = currentYear - 2;
+        }
+    }
+    if(financialYearMonth === 'January'){
+        selectedPeriod = selectedYear;
+        selectedStartDate = moment(selectedYear, 'YYYY').toDate();
+        selectedEndDate = moment(selectedYear, 'YYYY').endOf('year').toDate();
+    } else {
+        selectedPeriod = selectedYear + financialYearMonth;
+        selectedStartDate = moment(selectedPeriod, 'YYYYMMMM').toDate();
+        selectedEndDate = moment(selectedPeriod, 'YYYYMMMM').add(1,'years').subtract(1, 'months').endOf('month').toDate();
```

Comment:
how about `moment(selectedPeriod, 'YYYYMMMM').add(1,'years').subtract(1, 'days')`?

Reply:
![image](https://user-images.githubusercontent.com/43912578/103258418-e0d91580-49ba-11eb-9d01-78ec34480403.png)

Reply:
also you can say, `selectedStartDate.add(1,'years').subtract(1, 'days')` this is common for January and other months.

Reply:
nope it's not working

Reply:
`.toDate()`